components/audioinput/microphone/microphone.go

```diff
@@ -77,30 +77,38 @@ func newMicrophoneSource(attrs *Attrs, logger golog.Logger) (audioinput.AudioInp
 			return nil, err
 		}
 	}
-
-	labels := gostream.QueryAudioDeviceLabels()
-	for _, label := range labels {
-		if debug {
-			logger.Debugf("%s", label)
-		}
-
-		if pattern != nil && !pattern.MatchString(label) {
-			if debug {
-				logger.Debug("\t skipping because of pattern")
+	all := gostream.QueryAudioDevices()
+
+	for i, info := range all {
+		logger.Debugf("%s", info.ID)
+		logger.Debugf("\t labels: %v", info.Labels)
+		for _, p := range info.Properties {
+			logger.Debugf("\t %+v", p.Audio)
+			if pattern != nil && !pattern.MatchString(info.Labels[i]) {
+				if debug {
+					logger.Debug("\t skipping because of pattern")
+				}
+				continue
 			}
-			continue
-		}
-
-		s, err := tryMicrophoneOpen(label, gostream.DefaultConstraints, logger)
-		if err == nil {
-			if debug {
-				logger.Debug("\t USING")
+			if p.Audio.ChannelCount > 0 {
+				if pattern != nil && !pattern.MatchString(info.Labels[i]) {
+					if debug {
+						logger.Debug("\t skipping because of pattern")
+					}
+					continue
+				}
```

Comment:
you already did this check above- you can eliminate it.

Reply:
+1